src/library/scala/util/Using.scala

```diff
@@ -12,131 +12,202 @@
-    if (used.getAndSet(true)) throw new IllegalStateException("resource has already been used")
-    else Using.resource(resource)(f)
-}
+    /** Registers the specified resource with this manager, so that
+      * the resource is released when the manager is closed.
+      */
+    def acquire[R: Releasable](resource: R): Unit = {
+      if (resource == null) throw new NullPointerException("null resource")
+      if (closed) throw new IllegalStateException("Manager has already been closed")
+      resources = new Resource(resource) :: resources
+    }
 
-/** @define recommendUsing                   It is highly recommended to use the `Using` construct,
-  *                                          which safely wraps resource usage and management in a `Try`.
-  * @define multiResourceSuppressionBehavior If more than one exception is thrown by the operation and releasing resources,
-  *                                          the exception thrown ''first'' is thrown, with the other exceptions
-  *                                          [[java.lang.Throwable.addSuppressed(Throwable) added as suppressed exceptions]]
-  *                                          to the one thrown first. This is the case ''unless'' a later exception is
-  *                                          [[scala.util.control.NonFatal fatal]], and the one preceding it is not. In that case,
-  *                                          the first exception is added as a suppressed exception to the fatal one, and the fatal
-  *                                          one is thrown. If an exception is a
-  *                                          [[scala.util.control.ControlThrowable ControlThrowable]], no exception will be added to
-  *                                          it as a suppressed exception.
-  */
-object Using {
-  /** Creates a `Using` from the given resource.
-    *
-    * @note If the resource does not have an implicit [[Resource]] in
-    *       scope, the returned `Using` will be useless.
-    */
-  def apply[R](resource: => R): Using[R] = new Using(resource)
+    private def manage[A](op: Manager => A): A = {
+      var toThrow: Throwable = null
+      try {
+        op(this)
+      } catch {
+        case t: Throwable =>
+          toThrow = t
+          null.asInstanceOf[A] // compiler doesn't know `finally` will throw
+      } finally {
+        closed = true
+        var rs = resources
+        resources = null // allow GC, in case something is holding a reference to `this`
+        while (rs.nonEmpty) {
+          val resource = rs.head
+          rs = rs.tail
+          try resource.release()
+          catch {
+            case t: Throwable =>
+              if (toThrow == null) toThrow == t
+              else toThrow = preferentiallySuppress(toThrow, t)
+          }
+        }
+        if (toThrow != null) throw toThrow
+      }
+    }
+  }
+
+  object Manager {
```

Comment:
my thought was to save `Managed` for an `apply` method taking an implicit function, once those arrive, which I think makes syntax a bit neater; at that point, this `apply` method on `Manager` could be deprecated